plugins/octant/cmd/antrea-octant-plugin/traceflow.go

```diff
@@ -88,139 +95,167 @@ func getDstType(tf *crdv1alpha1.Traceflow) string {
 	return ""
 }
 
-// actionHandler handlers clicks and actions from "Start New Trace" and "Generate Trace Graph" buttons.
+// actionHandler handlers clicks and actions from "Start New Trace", "START NEW LIVE-TRAFFIC TRACE" and "Generate Trace Graph" buttons.
 func (p *antreaOctantPlugin) actionHandler(request *service.ActionRequest) error {
 	actionName, err := request.Payload.String("action")
 	if err != nil {
-		log.Printf("Failed to get input at string: %s", err)
+		log.Printf("Failed to get input at string: %s\n", err)
 		return nil
 	}
-
 	switch actionName {
-	case addTfAction:
-		srcNamespace, err := request.Payload.String(srcNamespaceCol)
-		if err != nil {
-			log.Printf("Invalid user input, CRD creation or Traceflow request may fail: "+
-				"failed to get srcNamespace as string: %s", err)
-			alert := action.CreateAlert(action.AlertTypeError, fmt.Sprintf("Failed to get source namespace as "+
-				"string: %s", err), action.DefaultAlertExpiration)
-			request.DashboardClient.SendAlert(request.Context(), request.ClientID, alert)
-			return nil
-		}
-		if errs := validation.ValidateNamespaceName(srcNamespace, false); len(errs) != 0 {
-			log.Printf("Invalid user input, CRD creation or Traceflow request may fail: "+
-				"failed to validate source namespace string %s, errs: %#v", srcNamespace, errs)
-			alert := action.CreateAlert(action.AlertTypeError, fmt.Sprintf("Invalid source namespace string, "+
-				"please check your input and submit again."), action.DefaultAlertExpiration)
-			request.DashboardClient.SendAlert(request.Context(), request.ClientID, alert)
-			return nil
+	case addTfAction, addLiveTfAction:
```

Comment:
I understand there can be some common code for these two cases, but I am still thinking if we should split these two cases to do validation. Because it seems that there are multiple boolean flags and goto clause here. What do you think?

Reply:
that make sense, I will check, maybe I can move more common codes as functions and make thess two actions into two cases which should help other to understand the logic better.